src/Microsoft.AspNetCore.Sockets/Transports/ServerSentEventsTransport.cs

```diff
@@ -37,6 +37,9 @@ public ServerSentEventsTransport(ReadableChannel<Message> application, ILoggerFa
 
             context.Response.Headers["Content-Encoding"] = "identity";
 
+            // Workaround for a Firefox bug where EventSource won't fire the open event
+            // until it receives some data
+            await context.Response.WriteAsync(":\r\n");
```

Comment:
Why would we add custom magic that fits neither event source protocol nor ours if we can use a built-in feature of a protocol we are using. Note that with custom init TS client parsers would have to be updated as well. Also, if you want to do this correctly you would have to keep the state to detect cases where you get multiple init messages. Oh, in that case should we also send init for longpolling and websockets? But if we do this for webSockets at the Sockets level the IoT scenario - being able to connect to the server with bare webSocket is broken because the user need to react to (ignore) the init message
We just added a new future across the stack that no one (at the moment) needs or asked. 

If we want to have an init message that has a purpose other then working around a FF bug we need to design it.